overlord/snapstate/check_snap.go

```diff
@@ -181,6 +181,11 @@ func checkCoreName(st *state.State, snapInfo, curInfo *snap.Info, flags Flags) e
 		return err
 	}
 
+	// FIMXE: make this more elegant
```

Comment:
It would be nice to document the original function and perhaps to expand on this comment as the purpose is somewhat elusive IMHO